src/store/tools/useAnnotationTool.ts

```diff
@@ -0,0 +1,162 @@
+import { Ref, computed, del, ref, set } from '@vue/composition-api';
+import { removeFromArray } from '@/src/utils';
+
+import { TOOL_COLORS } from '@/src/config';
+import { defineStore } from 'pinia';
+import { useCurrentImage } from '@/src/composables/useCurrentImage';
+import {
+  FrameOfReference,
+  frameOfReferenceToImageSliceAndAxis,
+} from '@/src/utils/frameOfReference';
+import { useViewStore } from '@/src/store/views';
+import { getLPSAxisFromDir } from '@/src/utils/lps';
+import { useViewConfigStore } from '@/src/store/view-configs';
+import { LPSAxisDir } from '@/src/types/lps';
+import { findImageID, getDataID } from '@/src/store/datasets';
+
+const useDependencyInjectionStore = defineStore('dependencyInjection', {});
+const useIDGenerator = () => {
+  const { $id } = useDependencyInjectionStore();
+  return $id;
+};
```

Comment:
I think we should pass `this` to `useAnnotationTool`, which will provide us with the injected dependencies.

Reply:
How do I get a reference to `this` in the "setup" function of a store?
```
export const useRulerStore = defineStore('ruler', () => {
  this // nope
```

Reply:
You can use `type _This = ReturnType<typeof useRulerStore>`, with `useAnnotationTool` using `Store` (named import from `pinia`) as the type.

Reply:
I'm missing how to do this.  `useAnnotationTool` is in the top level "setup" arrow function scope where `this` is `undefined`.

```
export const useRectangleStore = defineStore('rectangles', () => {
  const toolStoreProps = useAnnotationTool({
    services: this, // this undefined in arrow function
    toolDefaults: rectangleDefaults,
  });
```

Reply:
Oh, right! I forgot about the context. For function-based stores, `this` is only accessible on functions exposed on the store.

```
const useStore = defineStore('store', () => {
  type _This = Store // or type _This = ReturnType<typeof useStore>
  function generateID(this: _This) {
    return this.$id.nextID()
  }
  return {
    generateID,
  }
})
```

Reply:
Let's just switch to an ID store. I forget why I opted to have ID generation in an injected dependency, but it probably doesn't apply anymore.

You won't need to pass in the ID generator to `useAnnotationTool`. You'd just call `useIdStore` in `useAnnotationTool`.

Reply:
OK.  I added an IDStore and use it in composible.